ruby/vampires.rb

```diff
@@ -0,0 +1,128 @@
+# How many times do we need to fill out the questionnaire? 
+puts "How many employees are you processing right now?"
+employees = gets.chomp
+employees = employees.to_i.to_s.to_i
+while employees == 0 || employees == nil
+	puts "I'm sorry, could you try that again? Enter an integer greater than 0."
+	employees = gets.chomp
+	employees = employees.to_i.to_s.to_i
+end
+until employees > 0
+	puts "That's not a valid number. Try again."
+	employees = gets.chomp
+	employees = employees.to_i.to_s.to_i
+end
+
+def vamp_questionnaire 
+# Initial vampire status
+vamp_identity = "Results are inconclusive on whether this employee is a vampire."
+
+# Get a name
+puts "What is employee's name?"
+name = gets.chomp
+
+# Get employee age
+puts "How many years old are they?"
+age = gets.chomp
+age = age.to_i.to_s.to_i
+while age == 0 || age == nil
+	puts "I'm sorry, could you enter that as an integer?"
+	age = gets.chomp
+	age = age.to_i.to_s.to_i
+end
+
+# Get employee year of birth
+puts "What year were they born?"
+birth_year = gets.chomp
+birth_year = birth_year.to_i.to_s.to_i
+while birth_year == 0 || birth_year == nil 
+	puts "I'm sorry, could you enter that as an integer?"
+	birth_year = gets.chomp
+	birth_year = birth_year.to_i.to_s.to_i
+end
+
+# Does age match birth year?
+age_math = (age - (2017 - birth_year))
+if age_math == 0
+	age_math = true
+else age_math = false
+end
+
+# Ask if this person likes garlic bread
+puts "Our cafeteria has garlic bread. Would they like to eat some right now? (yes/no)"
+garlic_bread = gets.chomp
+until garlic_bread.downcase == "yes" || garlic_bread.downcase == "no"
+	puts "I'm sorry, give me a yes or no, please."
+	garlic_bread = gets.chomp
+end
+
+# Convert yes/no into boolean
+if garlic_bread.downcase == "yes"
+	garlic_bread = true 
+else garlic_bread = false 
+end
+
+
+# Ask if they want health insurance
+puts "We have a great healthcare plan. Should we sign them up for health insurance? (yes/no)"
+insurance = gets.chomp
+until insurance.downcase == "yes" || insurance.downcase == "no"
+	puts "I'm sorry, give me a yes or no, please."
+	insurance = gets.chomp
+end
+
+# Convert insurance into boolean
+if insurance.downcase != "yes"
+	insurance = false
+else insurance = true
+end
+
+#Clear allergies value before asking about it
+allergies = "none"
+
+# Ask about allergies
+puts "Please list any allergies you have. Hit return after each one. Enter 'Done' when done listing."
+allergies = gets.chomp
+until (allergies.downcase == "done") || (allergies.downcase == "sunshine")
+	allergies = gets.chomp
+end
+if allergies.downcase == "sunshine"
+	vamp_identity = "#{name} is probably a vampire."
+end
+
+# Override other outcomes based on sunshine allergy
+if vamp_identity != "#{name} is probably a vampire."
```

Comment:
I think you are on the right track here, but this logic could be simplified. It is not at its most clear representation yet.